pilot/pkg/serviceregistry/kube/controller/util.go

```diff
@@ -178,7 +178,7 @@ func getNodeSelectorsForService(svc *v1.Service) labels.Instance {
 	if nodeSelector := svc.Annotations[kube.NodeSelectorAnnotation]; nodeSelector != "" {
 		var nodeSelectorKV map[string]string
 		if err := json.Unmarshal([]byte(nodeSelector), &nodeSelectorKV); err != nil {
-			log.Debugf("failed to unmarshal node selector annotation value for service %s.%s: %v",
```

Comment:
it was intended, otherwise a invalid setting can bring spammy logs

Reply:
@j2gg0s maybe adding a metric?